src/cache.js

```diff
@@ -30,6 +30,25 @@ export function createCache({ allowReturnExpiredValue, expirationTime }) {
   }
 }
 
+/**
+ * @param {string | array} queryKey
+ * @returns string
+ */
+export function hashQueryKey(queryKey) {
+  const value = Array.isArray(queryKey) ? queryKey : [queryKey]
+
+  return JSON.stringify(value, (_, val) =>
+    val && typeof val === 'object' && !Array.isArray(val)
+      ? Object.keys(val)
+        .sort()
+        .reduce((result, key) => {
+          result[key] = val[key]
+          return result
+        }, {})
```

Comment:
Deze is niet nodig als je `Object.entries` gebruikt en daar `sort` op key.

In de basis is deze variant niet future proof, er is namelijk geen garantie dat deze techniek resulteert in een gesorteerd object. De meeste JS-engines hanteren nu wel het principe dat de volgorde van toevoegen telt, maar dat staan niet in de specificaties.